static/app/utils/profiling/hooks/useVirtualizedTree/useVirtualizedTree.tsx

```diff
@@ -242,72 +352,234 @@ export function useVirtualizedTree<T extends TreeLike>(
+    // In that case, check if the top position where the user clicked overlaps
+    // with a row and select that row.
+    const handleClick = (evt: MouseEvent) => {
+      if (evt.target !== scrollContainer) {
+        // user clicked on an element inside the container, defer to onClick
+        return;
+      }
+
+      const rect = (evt.target as HTMLDivElement).getBoundingClientRect();
+      const index = Math.floor(
+        (latestStateRef.current.scrollTop + evt.clientY - rect.top) / props.rowHeight
+      );
+
+      // If a node exists at the index, select it
+      if (tree.flattened[index]) {
+        dispatch({type: 'set tab index key', payload: index});
+        updateGhostRow({
+          ref: clickedGhostRowRef,
+          tabIndexKey: index,
+          scrollTop: latestStateRef.current.scrollTop,
+          rowHeight: props.rowHeight,
+          backgroundColor: theme.blue300,
+        });
+      }
+    };
+
+    // Because nodes dont span the full width, it's possible for users to
+    // click on a node at the far right end which is outside of the row width.
+    // In that case, check if the top position where the user clicked overlaps
+    // with a row and select that row.
+    const handleMouseMove = (evt: MouseEvent) => {
+      if (evt.target !== scrollContainer) {
+        // user clicked on an element inside the container, defer to onClick
+        return;
+      }
+
+      const rect = (evt.target as HTMLDivElement).getBoundingClientRect();
+      const index = Math.floor(
+        (latestStateRef.current.scrollTop + evt.clientY - rect.top) / props.rowHeight
+      );
+
+      // If a node exists at the index, select it
+      if (tree.flattened[index] && index !== latestStateRef.current.tabIndexKey) {
+        updateGhostRow({
+          ref: hoveredGhostRowRef,
+          tabIndexKey: index,
+          scrollTop: latestStateRef.current.scrollTop,
+          rowHeight: props.rowHeight,
+          backgroundColor: theme.blue100,
+        });
+      }
+    };
+
+    scrollContainer.addEventListener('click', handleClick);
+    scrollContainer.addEventListener('mousemove', handleMouseMove);
+
+    return () => {
+      scrollContainer.removeEventListener('click', handleClick);
+      scrollContainer.removeEventListener('mousemove', handleMouseMove);
+    };
+  }, [props.rowHeight, props.scrollContainer, tree.flattened]);
+
+  // When mouseleave is triggered on the contianer,
+  // we need to hide the ghost row to avoid an orphaned row
+  useEffect(() => {
+    const container = props.scrollContainer;
+    if (!container) {
+      return undefined;
+    }
+
+    function onMouseLeave() {
+      hideGhostRow({
+        ref: hoveredGhostRowRef,
+      });
+    }
+    container.addEventListener('mouseleave', onMouseLeave, {
+      passive: true,
+    });
+
+    return () => {
+      container.removeEventListener('scroll', onMouseLeave);
+    };
+  });
```

Comment:
I think you need a dep on the scroll container here.
```suggestion
  }, [props.scrollContainer]);
```